Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -333,6 +358,35 @@ void UPSWorldSubsystem::OnSpotComponentLoad(UPSSpotComponent* SpotComponent)
 	PSCurrentSpotComponentInternal = SpotComponent;
 }
 
+// Removes progression system spot that should not be available by other objects anymore.
+void UPSWorldSubsystem::RemoveProgressionSystemSpot(UPSSpotComponent* ProgressionSystemSpotComponent)
+{
+	if (ensureMsgf(ProgressionSystemSpotComponent, TEXT("%s: 'ProgressionSystemSpotComponent' is null"), *FString(__FUNCTION__)))
+	{
+		PSSpotComponentArrayInternal.RemoveSwap(ProgressionSystemSpotComponent);
+	}
+}
+
+// Destroy all star actors that should not be available by other objects anymore.
+void UPSWorldSubsystem::DestroyStarActors()
+{
+	UPoolManagerSubsystem::Get().ReturnToPoolArray(PoolActorHandlersInternal);
+	UPoolManagerSubsystem::Get().EmptyPool(UPSDataAsset::Get().GetStarActorClass());
+	PoolActorHandlersInternal.Empty();
+	ProgressionSettingsDataInternal.Empty();
+	StarDynamicProgressMaterial = nullptr;
+
+	PSDataAssetInternal.Reset();
+	PSHUDComponentInternal = nullptr;
+	PSSpotComponentArrayInternal.Empty();
+	PSCurrentSpotComponentInternal = nullptr;
+	if (SaveGameDataInternal)
+	{
+		SaveGameDataInternal->ConditionalBeginDestroy();
+		SaveGameDataInternal = nullptr;
+	}
+}
```

Comment:
`RemoveObserver` is missing, you only Add observer, but never remove. It's memory leak. Please remove observer from your main cleanup function, such as `PerformCleanup` from [comment](https://github.com/h4rdmol/ProgressionSystem/pull/26/files#r1818111332)

Reply:
@h4rdmol Looks good. Please test unload+load in the build with this code before we close this task. I don't trust `WITH_EDITOR` and suspect the Subsystem itself more than the Editor. Please let me know here if the PS module is able to load again after being unloaded

Reply:
@h4rdmol, I'm still waiting confirmation regarding this, as `!WITH_EDITOR` looks suspicious

Reply:
@JanSeliv no longer relevant as this functions completely removed.